karate-tests/src/test/java/mocks/sandbox/patch-patient.js

```diff
@@ -82,86 +87,85 @@ function patchPatient (originalPatient, request) {
   }
 
   const validOperations = ['add', 'replace', 'remove', 'test']
-  for (let i = 0; i < request.body.patches.length; i++) {
-    const patch = request.body.patches[i]
+  // Validate patch operations
+  for (const patch of request.body.patches) {
     if (!validOperations.includes(patch.op)) {
       return setInvalidValueError(`Invalid value - '${patch.op}' in field '0/op'`)
     }
   }
+
   const updatedPatient = JSON.parse(JSON.stringify(originalPatient))
-  let forbiddenUpdate = false
+  let forbiddenUpdate = null
   const updateErrors = []
 
-  for (let i = 0; i < request.body.patches.length; i++) {
-    const patch = request.body.patches[i]
-    if (patch.op === 'add' && patch.path === '/name/-') {
-      if (patch.value.use === 'usual') {
-        forbiddenUpdate = 'Forbidden update with error - multiple usual names cannot be added'
-      } else {
-        patch.value.id = 'new-object-id'
-        updatedPatient.name.push(patch.value)
-      }
-    }
-    if (patch.op === 'add' && patch.path === '/name/0/suffix') {
-      updatedPatient.name[0].suffix = patch.value
-    }
-    if (patch.op === 'add' && patch.path === '/name/0/suffix/0') {
-      updatedPatient.name[0].suffix.splice(0, 0, patch.value)
-    }
-    if (patch.op === 'replace' && patch.path === '/name/0/given/0') {
-      updatedPatient.name[0].given[0] = patch.value
-    }
-    if (patch.op === 'replace' && patch.path === '/gender') {
-      updatedPatient.gender = patch.value
-    }
-    if (patch.op === 'replace' && patch.path === '/birthDate') {
-      updatedPatient.birthDate = patch.value
-    }
-    if (patch.op === 'remove' && patch.path === '/name/0/suffix') {
-      if (!updatedPatient.name[0].suffix) {
-        updateErrors.push("Invalid update with error - Invalid patch - can't remove non-existent object '0'")
-      } else {
-        delete updatedPatient.name[0].suffix
-      }
+  const addNewName = (value) => {
+    if (value.use === 'usual') {
+      forbiddenUpdate = 'Forbidden update with error - multiple usual names cannot be added'
+    } else {
+      value.id = 'new-object-id'
+      updatedPatient.name.push(value)
     }
```

Comment:
Can we move the rest of these inline functions to the `operations.js`  file?

Reply:
just moved them to operations.js